fastly/block_fastly_service_v1_logging_loggly.go

```diff
@@ -0,0 +1,202 @@
+			},
+
+			"format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Apache-style string or VCL variables to use for log formatting.",
+			},
+
+			// Optional fields
+
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either 1 or 2. (default: 2)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+
+			"response_condition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the condition to apply",
+			},
+		},
+	},
+}
+
+func processLoggly(d *schema.ResourceData, conn *fst.Client, latestVersion int) error {
+	serviceID := d.Id()
+	ol, nl := d.GetChange("logging_loggly")
+
+	if ol == nil {
+		ol = new(schema.Set)
+	}
+	if nl == nil {
+		nl = new(schema.Set)
+	}
+
+	ols := ol.(*schema.Set)
+	nls := nl.(*schema.Set)
+
+	removeLogglyLogging := ols.Difference(nls).List()
+	addLogglyLogging := nls.Difference(ols).List()
+
+	// DELETE old Loggly logging endpoints
+	for _, oRaw := range removeLogglyLogging {
+		of := oRaw.(map[string]interface{})
+		opts := buildDeleteLoggly(of, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly Loggly logging endpoint removal opts: %#v", opts)
+
+		if err := deleteLoggly(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	// POST new/updated Loggly logging endpoints
+	for _, nRaw := range addLogglyLogging {
+		lf := nRaw.(map[string]interface{})
+		opts := buildCreateLoggly(lf, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly Loggly logging addition opts: %#v", opts)
+
+		if err := createLoggly(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readLoggly(conn *fst.Client, d *schema.ResourceData, s *fst.ServiceDetail) error {
+	// refresh Loggly
+	log.Printf("[DEBUG] Refreshing Loggly logging endpoints for (%s)", d.Id())
+	logglyList, err := conn.ListLoggly(&fst.ListLogglyInput{
+		Service: d.Id(),
+		Version: s.ActiveVersion.Number,
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERR] Error looking up Loggly logging endpoints for (%s), version (%v): %s", d.Id(), s.ActiveVersion.Number, err)
+	}
+
+	ell := flattenLoggly(logglyList)
+
+	if err := d.Set("logging_elasticsearch", ell); err != nil {
+		log.Printf("[WARN] Error setting Loggly logging endpoints for (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func createLoggly(conn *fst.Client, i *fst.CreateLogglyInput) error {
+	_, err := conn.CreateLoggly(i)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteLoggly(conn *fst.Client, i *fst.DeleteLogglyInput) error {
+	err := conn.DeleteLoggly(i)
+
+	if errRes, ok := err.(*fst.HTTPError); ok {
```

Comment:
Just noting that I pointed out in another PR that this would result in a 404 returning nil, right?  Is this expected?

Reply:
The 404 error handling is consistent w/ most of the existing client calls to delete objects (24 times in `fastly/resource_fastly_service_v1.go`.)